terminal/terminal.md

```diff
@@ -0,0 +1,417 @@
+     ls [-ABCFGHLOPRSTUW@abcdefghiklmnopqrstuwx1] [file ...]
+
+DESCRIPTION
+     For each operand that names a file of a type other than directory, ls displays its name as well as any requested, associated information.  For each operand that names a file of type directory, ls
+     displays the names of files contained within that directory, as well as any requested, associated information.
+
+     If no operands are given, the contents of the current directory are displayed.  If more than one operand is given, non-directory operands are displayed first; directory and non-directory operands
+     are sorted separately and in lexicographical order.
+```
+We can scroll down with the key arrows and quit **q**
+
+Whenever you don't know how to use a command use man then the name of the command to read everything about it, it also shows the flags you can use and in some cases some examples.
+
+But now what if only want a small description? Well we are all always in a hurry so let's be more pragmatic and use the **whatis** command
+
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ whatis ls
+ls (1)               - list directory contents
+```
+pretty neat
+
+What about the other way? I want a command if there is a command that lists the directory contents... if only...
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ apropos "list directory contents"
+```
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ apropos "list directory contents"
+dir (1)              - list directory contents
+ls (1)               - list directory contents
+ntfsls (8)           - list directory contents on an NTFS filesystem
+vdir (1)             - list directory contents
+```
+Yes, the command **apropos** searches for a command that fits your description
+
+#### Creating files and directories
+
+We have been exploring how to move across directories, list files and see the metadata of those, but now, let's go one step ahead and create a directory with the **mkdir** command
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir mydirectory
+danielvilla@Daniels-MacBook-Air:~$ 
+```
+Normally when a command is successful it won't print anything, if we look now the mydirectory directory is created and ready to have files
+
+It's very common that we want to create directories contained inside directories (that's how everything works) but mkdir alone won't let us do that
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir great_ideas/best_ideas/dontlookhere
+mkdir: great_ideas/best_deas: No such file or directory
+```
+We need to add the **p** flag to correct that
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir -p great_ideas/best_ideas/dontlookhere
+danielvilla@Daniels-MacBook-Air:~$ 
+```
+That will save us a lot of time of doing `mkdir great_ideas` then `cd great_ideas` then `mkdir best_ideas`... well you get the point
+
+NOW! we have our directory ready to have files and great ideas, but how do we create a simple file? 
+
+There are plenty of ways to create files on unix system but the common ones are
```

Comment:
There are plenty of ways to create files ~on~ `in` unix system but the common ones are